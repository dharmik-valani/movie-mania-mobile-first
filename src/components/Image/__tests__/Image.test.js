import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Imagecomponent from "../Image.component.jsx";

describe("Imagecomponent", () => {
  it("renders an image with the correct attributes", () => {
    const src = "test-image.jpg";
    const alttext = "Test Alt Text";
    const imageclassName = "test-class";
    const onClick = jest.fn();

    render(
      <Imagecomponent
        src={src}
        alttext={alttext}
        imageclassName={imageclassName}
        onClick={onClick}
      />
    );

    const imgElement = screen.getByAltText(alttext);
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src", src);
    expect(imgElement).toHaveClass(imageclassName);
  });

  test("matches image snapshot", () => {
    const src = "test-image.jpg";
    const alttext = "Test Alt Text";
    const imageclassName = "test-class";
    const onClick = jest.fn();

    const tree = renderer
      .create(
        <Imagecomponent
          src={src}
          alttext={alttext}
          imageclassName={imageclassName}
          onClick={onClick}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
