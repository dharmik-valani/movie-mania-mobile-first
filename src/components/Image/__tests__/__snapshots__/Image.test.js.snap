// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Imagecomponent matches image snapshot 1`] = `
<img
  alt="Test Alt Text"
  className="test-class"
  onClick={[MockFunction]}
  src="test-image.jpg"
/>
`;
